nilearn/surface/surface.py

```diff
@@ -1242,15 +1245,15 @@ class PolyData:
 
     Parameters
     ----------
-    left : :obj:`numpy.ndarray` or :obj:`str` or :obj:`pathlib.Path` or None,\
-           default = None
+    left : 2D :obj:`numpy.ndarray` or :obj:`str` or :obj:`pathlib.Path` \
+           or None, default = None
 
-    right : :obj:`numpy.ndarray` or :obj:`str` or :obj:`pathlib.Path` or None,\
-           default = None
+    right : 2D :obj:`numpy.ndarray` or :obj:`str` or :obj:`pathlib.Path` \
+            or None, default = None
```

Comment:
Something that could be nicer for users, is that if a 1D numpy array then it is converted to a 2D array.

@man-shu @bthirion WDYT?

Reply:
I see, so this would ensure that the data arrays are 2D, even if there is only one sample/timepoint.

1. Could cause some test failures on #4770. So we should merge this one first.
2. Would the saving/writing be unaffected? For example, would the .gii files saved by nilearn still work with other software?

Reply:
Otherwise I am ok with the changes.

Reply:
> 1. Could cause some test failures on

Yeah I was having a look at what you were doing and thought it may cause issues.

Reply:
> I see, so this would ensure that the data arrays are 2D, even if there is only one sample/timepoint.

Yes but I think it may be easier to use in general if we can still instantiate a Surface image with 1 dim numpy array for one hemisphere but we just force them to be 2 dim.

Reply:
Will wait for the test to all pass before I refactor things.